### PR TITLE
Don't assume build isn't None in a docker build env

### PR DESCRIPTION
**Pull request by stsewd**
This is kind of the same we hit with a local env
https://github.com/readthedocs/readthedocs.org/pull/6503

### CODE REVIEW

File: readthedocs/doc_builder/environments.py [hunk 1]
```diff
@@ -792,7 +792,7 @@ def __init__(self, *args, **kwargs):
         self.container = None
         self.container_name = slugify(
             'build-{build}-project-{project_id}-{project_name}'.format(
-                build=self.build.get('id'),
+                build=self.build.get('id') if self.build else '',
```
Comment (stsewd) [inline readthedocs/doc_builder/environments.py]: The name may collapse if two tasks are triggered in the same lapse of time. Not sure what happens there.

Comment (agjohnson) [inline readthedocs/doc_builder/environments.py]: More clear would be container names of:

* `build-1234-project-123-foo`
* `sync-project-123-foo`

File: readthedocs/doc_builder/environments.py [hunk 2]
```diff
@@ -931,7 +931,7 @@ def get_client(self):
             # Instead, give the user a generic failure
             raise BuildEnvironmentError(
                 BuildEnvironmentError.GENERIC_WITH_BUILD_ID.format(
-                    build_id=self.build['id'],
+                    build_id=self.build.get('id') if self.build else '',
```
Comment (agjohnson) [inline readthedocs/doc_builder/environments.py]: I feel like this whole block should be conditional on whether or not this is a build operation vs sync operation. Is there any UI to even show this error if raised in a sync operation?

Comment (stsewd) [reply readthedocs/doc_builder/environments.py]: Nope. Yeah, we can just put a conditional

File: readthedocs/doc_builder/environments.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/doc_builder/environments.py]: Just to answer my own question, we do a check for this in the __enter__ event.

**Pull request merged**